src/macaron/slsa_analyzer/pypi_heuristics/metadata/unchanged_release.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2024 - 2024, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""Heuristic analyzer to check unchanged content in multiple releases."""
+
+from collections import Counter
+
+from macaron.slsa_analyzer.package_registry.pypi_registry import PyPIApiClient
+from macaron.slsa_analyzer.pypi_heuristics.analysis_result import RESULT
+from macaron.slsa_analyzer.pypi_heuristics.base_analyzer import BaseAnalyzer
+from macaron.slsa_analyzer.pypi_heuristics.heuristics import HEURISTIC
+
+
+class UnchangedReleaseAnalyzer(BaseAnalyzer):
+    """Analyzer checks heuristic."""
+
+    def __init__(self, api_client: PyPIApiClient) -> None:
+        super().__init__(
+            name="unchanged_release_analyzer",
+            heuristic=HEURISTIC.UNCHANGED_RELEASE,
+            depends_on=[(HEURISTIC.HIGH_RELEASE_FREQUENCY, RESULT.FAIL)],  # Analyzing when this heuristic fail
+        )
+        self.hash: str = "sha256"
+        self.api_client = api_client
+
+    def _get_digests(self) -> list | None:
+        """Get all digests of the releases.
+
+        Returns
+        -------
+            list | None: Digests.
+        """
+        releases: dict | None = self.api_client.get_releases()
+        if releases is None:
+            return None
+        digests: list = [
+            metadata[0].get("digests").get(self.hash)
```

Comment:
Please use `json_extract` for extracting data from `releases` and use a normal for loop instead of list comprehension. 
I know that list comprehension works well and doesn't have any issue. However, it adds a bit of complexity to code maintaining and readability. Writing it in a traditional for loop make your logic explicit, which can be easily picked up by other people.

Reply:
Fixed this in 1d801c45b3f13c3cdb3e603935c314b1687cebc3